compiler/rustc_mir_build/src/thir/pattern/check_match.rs

```diff
@@ -405,7 +405,10 @@ impl<'p, 'tcx> MatchVisitor<'p, 'tcx> {
         scrut_ty: Ty<'tcx>,
     ) -> Result<UsefulnessReport<'p, 'tcx>, ErrorGuaranteed> {
         let pattern_complexity_limit =
-            get_limit_size(cx.tcx.hir().krate_attrs(), cx.tcx.sess, sym::pattern_complexity);
+            get_limit_size(cx.tcx.hir().krate_attrs(), cx.tcx.sess, sym::pattern_complexity)
+                // Default value to emit the warning for "too complex" match. We picked it based on
+                // this crater run: <https://github.com/rust-lang/rust/pull/121979>.
+                .or(Some(10_000_000));
```

Comment:
```suggestion
                // Default value to emit the warning for "too complex" match. We picked it to warn after a second or two.
                .or(Some(10_000_000));
```
I don't think the crater matters much to future readers of this code, but the timing intent does.

Reply:
Actually you said that limit took more like 20s on your machine? I'd be curious to crater a lower limit then, because I'm afraid people won't wait 20s

Reply:
Fine by me!